src/test/java/integrationtest/CompanyRestApiIntegrationTest.java

```diff
@@ -247,4 +247,801 @@ public class CompanyRestApiIntegrationTest {
         """, timeStamp), response, new CustomComparator(JSONCompareMode.STRICT, new Customization("timestamp", (((o1, o2) -> true
         )))));
     }
+
+    @Test
+    @DataSet(value = "datasets/companies.yml")
+    @Transactional
+    void 企業登録ができること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/companies")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content("""
+                        {
+                            "companyName": "未登録株式会社",
+                            "companyPhone": "0312345678",
+                            "region": "神奈川県",
+                            "city": "川崎市",
+                            "address": "高津区1-1-1",
+                            "companyRank": "S",
+                            "salesPersonId": "1"
+                        }
+                        """))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        JSONAssert.assertEquals("""
+        {
+                "message": "企業が正常に登録されました。",
+                "ID": 5
+        }
+        """, response, new CustomComparator(JSONCompareMode.STRICT,
+                new Customization("ID", (((o1, o2) -> true)))));
+    }
+
+    @Test
+    @DataSet(value = "datasets/companies.yml")
+    @Transactional
+    void 企業登録のリクエストで企業名がnullの時に例外がスローされること() throws Exception {
```

Comment:
例外がスローされるというよりは、バリデーションエラーになるといった方が正しいかと思いました。他のテストケース名も同様です！
```suggestion
    void 企業登録のリクエストで企業名がnullの時にバリデーションエラーとなること() throws Exception {
```